--- sxalert.c	2023-11-18 13:24:57.322378330 +0000
+++ sxalertxr.c	2023-11-18 13:24:26.619213478 +0000
@@ -70,6 +70,40 @@
 	return max;
 }
 
+void
+get_xresources_colors(void)
+{
+	FILE *f = fopen(xresources, "r");
+	if (!f) {
+		fprintf(stderr, "Could not open file: %s\n", xresources);
+		exit(EXIT_FAILURE);
+	}
+
+	char buffer[XR_BUFFER];
+
+	char colors[2][8];
+	int clr_i=0;
+
+	char *arr[2];
+
+	for (int i=0; i<4; i++) {
+		fgets(buffer, sizeof(buffer), f);
+
+		if (i <= 1) continue;
+
+		arr[0] = strtok(buffer, "#");
+		arr[1] = strtok(NULL, "#\n");
+
+		strncpy(colors[clr_i++], arr[1], 7);
+	}
+
+	strcpy(text_color, colors[0]);
+	strcpy(border_color, colors[0]);
+	strcpy(bg_color, colors[1]);
+
+	fclose(f);
+}
+
 static int
 get_width(Display *dpy, XftFont *font, char **lines, int length)
 {
@@ -176,6 +210,8 @@
 	char text[BUFFER];
 	extern char *optarg;
 
+	get_xresources_colors();
+
 	while ((c = getopt(argc, argv, "d:b:t:g:r:vh")) != -1 ) {
 		switch (c) {
 		case 'v':
--- config.h	2023-11-18 13:25:05.212335034 +0000
+++ configxr.h	2023-11-18 13:19:15.637586640 +0000
@@ -6,6 +6,7 @@
  */
 
 #define BUFFER 2054
+#define XR_BUFFER 128
 
 static int duration = 5000; /* in milliseconds */
 
@@ -26,3 +27,5 @@
 
 const char *fontname = "DejaVu Sans Mono:size=13:antialias=true";
 static int text_height = 11;
+
+const char *xresources = "/home/$USER/.cache/wal/colors.Xresources"; /* wal */
